core/string/ustring.cpp

```diff
@@ -3368,7 +3368,7 @@ int String::find(const char *p_str, int p_from) const {
 	return -1;
 }
 
-int String::find_char(const char32_t &p_char, int p_from) const {
+int String::find_char(char32_t p_char, int p_from) const {
```

Comment:
char32_t is just 4 bytes so small enough that passing by value is more efficient.

Reply:
Is this still true when `p_char` is passed to `_cowdata.find` which is `find(const T &p_val, Size p_from) const`?

Reply:
Yeah good question. So, the internals of the function won't affect how the parameter is passed so there'd always be a ref passed. Can play in this compiler explorer to see this: https://godbolt.org/z/Pq6e6McGG

Adding the ref, removing the ref generates different assembly, no matter the internals of the function. And it's just a little more costly to send a ref of small types.